woo-razorpay.php

```diff
@@ -1099,15 +1099,25 @@ function check_razorpay_response()
 
             rzpLogInfo("Called check_razorpay_response: $post_password");
 
-            $postData = $wpdb->get_results( $wpdb->prepare("SELECT ID, post_status FROM $wpdb->posts AS P WHERE post_type=%s AND post_password = %s", $post_type, $post_password ) );
+            // $postData = $wpdb->get_results( $wpdb->prepare("SELECT ID, post_status FROM $wpdb->posts AS P WHERE post_type=%s AND post_password = %s", $post_type, $post_password ) );
 
-            $arrayPost = json_decode(json_encode($postData), true);
+            // if($postData == null)
+            // {
+                $meta_key = '_order_key';
+                $postData = $wpdb->get_row( $wpdb->prepare("SELECT post_id FROM $wpdb->postmeta AS P WHERE meta_key = %s AND meta_value = %s", $meta_key, $post_password ) );
+            // }
+            
 
-            if (!empty($arrayPost) && count($arrayPost[0]) > 0)
+            $postStatus = $wpdb->get_row( $wpdb->prepare("SELECT post_status FROM $wpdb->posts AS P WHERE post_type=%s and ID=%s", $post_type, $postData->post_id) );
+            
+            $arrayPost = json_decode(json_encode($postData), true);
+            if (!empty($arrayPost) && $arrayPost != null)
```

Comment:
@garimaprachi use and instead of && and every condition should start with new line